conda_forge_tick/migrators/core.py

```diff
@@ -0,0 +1,861 @@
+                ):
+                    inner_text = text.replace(a, b).replace(c, d)
+                    rendered = parse_meta_yaml(render_meta_yaml(inner_text))
+                    if requests.get(rendered["source"]["url"]).status_code == 200:
+                        text = inner_text
+                        # The above clauses could do bad things the version
+                        # itself
+                        text = text.replace("version: v{{ v", "version: {{ v")
+                        with open("meta.yaml", "w") as fp:
+                            fp.write(text)
+                        break
+        # Get patterns to replace checksum for each platform
+        rendered_text = render_meta_yaml(text)
+        urls = self.find_urls(rendered_text)
+        new_patterns = self.get_hash_patterns("meta.yaml", urls, hash_type)
+
+        with indir(recipe_dir):
+            for f, p, n in new_patterns:
+                p = eval_version(p)
+                n = eval_version(n)
+                replace_in_file(p, n, f)
+            self.set_build_number("meta.yaml")
+        return super().migrate(recipe_dir, attrs)
+
+    def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
+        pred = [
+            (name, self.ctx.effective_graph.nodes[name]["payload"]["new_version"])
+            for name in list(
+                self.ctx.effective_graph.predecessors(feedstock_ctx.package_name),
+            )
+        ]
+        body = super().pr_body(feedstock_ctx)
+        # TODO: note that the closing logic needs to be modified when we
+        #  issue PRs into other branches for backports
+        open_version_prs = [
+                    muid['PR']
+                    for muid in feedstock_ctx.attrs.get("PRed", [])
+                    if muid["data"].get("migrator_name") == "Version"
+                    # The PR is the actual PR itself
+                    and muid.get("PR", {}).get("state", None) == "open"
+                ],
+        muid: dict
+        body = body.format(
+            "It is very likely that the current package version for this "
+            "feedstock is out of date.\n"
+            "Notes for merging this PR:\n"
+            "1. Feel free to push to the bot's branch to update this PR if needed.\n"
+            "2. The bot will almost always only open one PR per version.\n"
+            "Checklist before merging this PR:\n"
+            "- [ ] Dependencies have been updated if changed\n"
+            "- [ ] Tests have passed \n"
+            "- [ ] Updated license if changed and `license_file` is packaged \n"
+            "\n"
+            "Note that the bot will stop issuing PRs if more than {} Version bump PRs "
+            "generated by the bot are open. If you don't want to package a particular "
+            "version please close the PR.\n\n"
+            "{}".format(self.max_num_prs, '\n'.join([f"Closes: #{muid['number']}" for muid in open_version_prs])),
```

Comment:
This will allow us to close out older version PRs, helping to keep things off the `could_use_help` list, reducing the bot's number of PRs to check/keep track of

Reply:
I support this, but we should make sure the bot is not autodeleting old branches that people have pushed to with work they want to keep.

Reply:
Hmm, would those branches still have value? Once a newer PR goes in will conda go back and build the old versions (if they are PRed)?

One of the issues here is that we can only issue this statement at PR open time, if someone goes and edits one of the older PRs after this PR is issued there is nothing we can do.

Reply:
To some extent maintainers should have their own local fork of the bot's branch so they could make their own PR in theory, although this assumes that they didn't directly edit on the webpage.

Reply:
At least when I am working on version updates, I sometimes fix other things in the PR as I go that are unrelated. I could imagine leaving changes in another branch that I'd want to come back to later, but that is just me personally. 

I am not worried about corner cases here in terms of people changing the bot comments etc.

Reply:
We could take the hardline that any commits in the bots branch are automanaged and could be deleted at any time.

Reply:
We might want to add a note to the bot PRs saying this so people are aware.

Reply:
We could also tighten up the bot's deletion logic to check if 
1. the PR was closed but not merged
1. the PR has non bot committers

If both are true don't delete the branch

Reply:
Unfortunately the auto closing done by github via the markdown link won't obey this logic, but presumably that is on the person that merged the PR.

Reply:
It seems I'm incorrect, we don't delete branches if they weren't merged.

https://github.com/regro/cf-scripts/pull/701/files#diff-edf8107a414f0eb7fa4f03ec48c9e2e6R231

Reply:
for example: https://github.com/conda-forge/pyutilib-feedstock/pull/31